libpod/container.go

```diff
@@ -407,6 +409,58 @@ func (c *Container) teardownStorage() error {
 	return nil
 }
 
+func (c *Container) addImageData(g *generate.Generator) (err error) {
+	var (
+		args []string
+	)
+	if !c.config.UseImageConfig {
+		return nil
+	}
+	imageData, err := c.runtime.GetImageConfig(c.config.RootfsImageID)
+	if err != nil {
+		return errors.Wrapf(err, "error reading container image data")
+	}
+	if imageData.User != "" {
+		uid, _, err := chrootuser.LookupUserInContainer(c.bundlePath(), imageData.User)
+		if err != nil {
+			return errors.Wrapf(err, "user %q is not defined in %q", imageData.User, c.bundlePath())
+		}
+		g.SetProcessUID(uint32(uid))
+	}
+	for _, e := range imageData.ExposedPorts {
```

Comment:
I don't know how Docker handles mixing image ports and user-specified port mappings - do we override the image ports when the user specifies them, or do we add the ones the user specified to the ones in the image (overriding only when port numbers match)?

Reply:
I would figure it is additive, but we need to check,  I am not sure if this is even handled in CRI-O.